### ISSUE TITLE
add timer for exercises

### ISSUE THREAD

**Issue by oliverglanz** [EzerIT/BibleOL#34]
Teachers have been asking whether it is possible to put a timer on exercises, i.e. give students a set time duration for an exercise so that they will not use grammars and dictionaries while taking an exercise.
For example, one could allow for a vocab exercise of 10 words a 3min duration. If the students has not finished the exercise within these 3min the exercise will automatically be submitted with "grade task".

**Comment by ernstboogert** [comment]
Yes, agreed... Timer should be optional and needs to be selected during the exercise creation procedure...

**Comment by tmccormack165** [comment]
**Frontend Design:**
<img width="721" alt="Screenshot 2024-03-07 at 1 11 48 PM" src="https://github.com/EzerIT/BibleOL/assets/60988784/4f0bc82c-2639-4642-8188-2861a08c33e6">

**Comment by tmccormack165** [comment]
Frontend Design on Edit Quiz Panel:
<img width="706" alt="Screenshot 2024-03-07 at 1 16 52 PM" src="https://github.com/EzerIT/BibleOL/assets/60988784/4e377378-cc8d-435f-8f6c-c80a43064cc4">

**Comment by oliverglanz** [comment]
Great!

**Comment by tmccormack165** [comment]
New Exercise Panel:
<img width="1246" alt="Screenshot 2024-03-08 at 9 39 45 PM" src="https://github.com/EzerIT/BibleOL/assets/60988784/b6426dae-6fc9-4fe0-9336-775b74b272fa">

When there is ten or less seconds remaining, the text turns red:
<img width="1238" alt="Screenshot 2024-03-08 at 9 39 58 PM" src="https://github.com/EzerIT/BibleOL/assets/60988784/132e4d29-88ca-4d03-bb10-989441422475">

**Comment by oliverglanz** [comment]
This looks good. I would suggest, however, that the switching to the red color happens when we hit 60 or lex seconds. This should be a more meaningful shift since students can still control their behavior when 60s are left. When 10s are left things are pretty much settled when it comes to outcomes.

**Comment by oliverglanz** [comment]
@tmccormack165, when do you think this will be implemented?

**Comment by tmccormack165** [comment]
@oliverglanz Hi Oliver, the goal is to have this done by next Wednesday 04/10
